content/docs/pipelines/pipelines-tutorial.md

```diff
@@ -0,0 +1,616 @@
+    the probability that the image matches the respective label.
+
+## Understanding the pipeline definition code
+
+The pipeline is defined in the Python file `mnist_pipeline.py` which you
+downloaded from GitHub. The following sections give an overview of the content
+of that file.
+
+### Decorator
+
+The `@dsl.pipeline` decorator provides metadata about the pipeline:
+
+```
+@dsl.pipeline(
+  name='MNIST',
+  description='A pipeline to train and serve the MNIST example.'
+)
+```
+
+### Function header
+
+The `mnist_pipeline` function defines the pipeline. The function includes a 
+number of arguments which are exposed in the Kubeflow Pipelines UI when you
+create a new run of the pipeline.
+Although you pass these argeuments as strings, the arguments are of type 
+[`kfp.dsl.PipelineParam`](https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/dsl/_pipeline_param.py).
+
+```
+def mnist_pipeline(model_export_dir='gs://your-bucket/export',
+                   train_steps='200',
+                   learning_rate='0.01',
+                   batch_size='100'):
+```
+
+### The training component (`train`)
+
+The following block defines the `train` component, which handles the training
+of the ML model: 
+
+```
+train = dsl.ContainerOp(
+    name='train',
+    image='gcr.io/kubeflow-examples/mnist/model:v20190304-v0.2-176-g15d997b',
+    arguments=[
+        "/opt/model.py",
+        "--tf-export-dir", model_export_dir,
+        "--tf-train-steps", train_steps,
+        "--tf-batch-size", batch_size,
+        "--tf-learning-rate", learning_rate
+        ]
+).apply(gcp.use_gcp_secret('user-gcp-sa'))
+```
+A component consists of a 
+[`kfp.dsl.ContainerOp`](https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/dsl/_container_op.py)
+object with a name and a container path. The container image for
+the MNIST training component is defined in the MNIST example's
+[`Dockerfile.model`](https://github.com/kubeflow/examples/blob/master/mnist/Dockerfile.model).
+
+The training component runs with access to your `user-gcp-sa` secret, which
+ensures the component has read/write access to your Cloud Storage bucket for
+storing the output from the model training.
+
+### The model serving component (`serve`)
+
+The following block defines the `serve` component, which serves the trained 
+model for prediction:
+
+```
+serve = dsl.ContainerOp(
+    name='serve',
+    image='gcr.io/ml-pipeline/ml-pipeline-kubeflow-deployer:\
+            7775692adf28d6f79098e76e839986c9ee55dd61',
+    arguments=[
+        '--model-export-path', model_export_dir,
+        '--server-name', "mnist-service"
+    ]
+).apply(gcp.use_gcp_secret('user-gcp-sa'))
+serve.after(train)
+```
+The `serve` component differs from the `train` component with respect to
+how long the service lasts. While `train` runs a single container and then 
+exits, `serve` runs a container that launches long-lived resources in the 
+cluster. 
```

Comment:
You might want to mention that you can see the long-lived resources running in the cluster with `kubectl get all | grep mnist-serve` (that's off the top of my head, you may way to double check)

Reply:
I'll leave this one for now, as I've torn down my environment and I don't want to put untested code in the tutorial. Please raise an issue if you think it'd be a useful addition.